code/modules/atmospherics/machinery/pipes/heat_exchange/junction.dm

```diff
@@ -32,6 +32,7 @@
 
 /obj/machinery/atmospherics/pipe/heat_exchanging/junction/update_icon_state()
 	icon_state = "pipe[nodes[1] ? "1" : "0"][nodes[2] ? "1" : "0"]-[piping_layer]"
+	update_layer()
```

Comment:
Assuming everything's getting called properly this shouldn't be needed, right? Since `/obj/machinery/atmospherics/pipe/update_icon()` calls `update_layer()`

Reply:
without the update layer
![image](https://user-images.githubusercontent.com/42839747/109486269-f0d8a880-7a82-11eb-9d83-4a3edee570a6.png)

with the update layer 
![image](https://user-images.githubusercontent.com/42839747/109486231-e0c0c900-7a82-11eb-9342-0532d7a8e930.png)

As you can see without having update_layer after the overlays are setted results in graphic issues.
So the update_layer needs to be done after

Reply:
I think the `update_layer` call isn't doing anything. The manifolds appear to be getting a copy of the center overlay hucked into their overlays list and then somehow ejected from the managed overlays list. I'm trying to track down where this is happening, but I'm not having much luck.

Reply:
the update_layer doesn't fix the "shadow" pipe, i'm sorry if i wasn't clear enough, what it does is fix the layer that the pipe is displayed on so that it removes those artifacts you see on the pipeline on the image above, as you see the image above has weird details if you go along the pipe, while is clear on the image below